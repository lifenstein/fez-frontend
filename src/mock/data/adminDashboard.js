export const adminDashboardConfig = {
    admin_users: [
        {id: 123, name: 'Michael Feeney'},
        {id: 23, name: 'Another Staff'},
        {id: 444, name: 'Elizabeth Alvey'},
        {id: 2333, name: 'Lee Sibbald'},
    ],
};

export const adminDashboardToday = {
    systemalerts: {
        total: 150,
        today: 25,
        assigned: 15,
        unassigned: 135,
        mine: 8,
    },
    works: {
        processed: 82,
        unprocessed: 15,
    },
    oa: {
        current: 256,
        total: 1256,
    },
};

export const adminDashboardQuickLinks =  [
    { qlk_id: 1, qlk_amount: 10, qlk_order: 1, qlk_title: '2021+ Imported Records with an Author ID and Research Subtypes Only', qlk_link: 'https://espace.library.uq.edu.au/records/search?searchQueryParams%5ball%5d=%28author_id:%5b1+TO+%2a%5d%29+AND+%28date:%5b2021%5c-01%5c-01T00%5c:00%5c:00Z+TO+2024%5c-12%5c-31T00%5c:00%5c:00Z+%5d%29+AND+%28subtype:%28%22Article%22%29+OR+subtype:%28%22Critical%22%29+OR+subtype:%28%22Fully%22%29+OR+subtype:%28%22Research+Book%22%29+OR+subtype:%28%22Research+Book+Chapter%22%29%29+AND+%28%28isi_loc:%5b%2a+TO+%2a%5d%29+OR+%28scopus_id:%5b%2a+TO+%2a%5d%29%29+AND+%28%28ismemberof:+UQ:639325%29+OR+%28ismemberof:+UQ:308300%29+OR+%28ismemberof:+UQ:180159%29+OR+%28ismemberof:+UQ:688533%29+OR+%28ismemberof:+UQ:727866%29%29&page=1&pageSize=20&sortBy=score&sortDirection=Desc' },
    { qlk_id: 7, qlk_amount: null, qlk_order: 6, qlk_title: 'UQ Library', qlk_link: 'https://www.library.uq.edu.au' },
    { qlk_id: 2, qlk_amount: 30, qlk_order: 7, qlk_title: '2022+ Imported Records with no Author ID with subtype exclusions', qlk_link: 'https://espace.library.uq.edu.au/records/search?searchQueryParams%5ball%5d=%28%21author_id:%5b1+TO+%2a%5d%29+AND+%28date:%5b2022%5c-01%5c-01T00%5c:00%5c:00Z+TO+2024%5c-12%5c-31T00%5c:00%5c:00Z%5d%29+AND+%28%21subtype:%28%22letter%22%29+OR+%21subtype:%28%22published+abstract%22%29+OR+%21subtype:%28%22review+of+book%22%29+OR+%21subtype:%28%22introduction%22%29+OR+%21subtype:%28%22poster%22%29%29+%28%28ismemberof:+UQ:639325%29+OR+%28ismemberof:+UQ:308300%29+OR+%28ismemberof:+UQ:180159%29+OR+%28ismemberof:+UQ:688533%29+OR+%28ismemberof:+UQ:727866%29%29&page=1&pageSize=20&sortBy=score&sortDirection=Desc' },
    { qlk_id: 3, qlk_amount: 60, qlk_order: 2, qlk_title: '2020 - 2023 Imported Records with an Author ID and Research Subtypes Only', qlk_link: 'https://espace.library.uq.edu.au/records/search?searchQueryParams%5ball%5d=%28author_id:%5b1+TO+%2a%5d%29+AND+%28date:%5b2020%5c-01%5c-01T00%5c:00%5c:00Z+TO+2023%5c-12%5c-31T00%5c:00%5c:00Z+%5d%29+AND+%28subtype:%28%22Article%22%29+OR+subtype:%28%22Critical%22%29+OR+subtype:%28%22Fully%22%29+OR+subtype:%28%22Research+Book%22%29+OR+subtype:%28%22Research+Book+Chapter%22%29%29+AND+%28%28isi_loc:%5b%2a+TO+%2a%5d%29+OR+%28scopus_id:%5b%2a+TO+%2a%5d%29%29+AND+%28%28ismemberof:+UQ:639325%29+OR+%28ismemberof:+UQ:308300%29+OR+%28ismemberof:+UQ:180159%29+OR+%28ismemberof:+UQ:688533%29+OR+%28ismemberof:+UQ:727866%29%29&page=1&pageSize=20&sortBy=score&sortDirection=Desc' },
    { qlk_id: 4, qlk_amount: 90, qlk_order: 3, qlk_title: '2021 - 2023 Imported Records with no Author ID with subtype exclusions', qlk_link: 'https://espace.library.uq.edu.au/records/search?searchQueryParams%5ball%5d=%28%21author_id:%5b1+TO+%2a%5d%29+AND+%28date:%5b2021%5c-01%5c-01T00%5c:00%5c:00Z+TO+2023%5c-12%5c-31T00%5c:00%5c:00Z%5d%29+AND+%28%21subtype:%28%22letter%22%29+OR+%21subtype:%28%22published+abstract%22%29+OR+%21subtype:%28%22review+of+book%22%29+OR+%21subtype:%28%22introduction%22%29+OR+%21subtype:%28%22poster%22%29%29+%28%28ismemberof:+UQ:639325%29+OR+%28ismemberof:+UQ:308300%29+OR+%28ismemberof:+UQ:180159%29+OR+%28ismemberof:+UQ:688533%29+OR+%28ismemberof:+UQ:727866%29%29+AND+%28%21institutional_status:%22Non-UQ%22%29&page=1&pageSize=20&sortBy=score&sortDirection=Desc' },
    { qlk_id: 5, qlk_amount: 110, qlk_order: 4, qlk_title: 'Not yet publicly available (with merged metadata)', qlk_link: 'https://espace.library.uq.edu.au/records/search?searchQueryParams%5ball%5d=%28isMemberOf:+UQ:676357%29+AND+%28%28doi:%5b%2a+TO+%2a%5d%29+OR+%28isi_loc:%5b%2a+TO+%2a%5d%29+OR+%28scopus_id:%5b%2a+TO+%2a%5d%29%29&page=1&pageSize=20&sortBy=score&sortDirection=Desc' },
    { qlk_id: 6, qlk_amount: 2345, qlk_order: 5, qlk_title: 'eSpace Followup', qlk_link: 'https://espace.library.uq.edu.au/records/search?page=1&pageSize=20&sortBy=score&sortDirection=Desc&searchQueryParams%5ball%5d%5bvalue%5d=ismemberof:+UQ:237927&searchQueryParams%5ball%5d%5blabel%5d=&searchMode=advanced' },
];

export const adminDashboardSystemAlerts = [
    {sat_id:1, sat_title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: undefined, sat_resolved_by: undefined },
    {sat_id:12, sat_title: 'Issues on record - UQ:34555c6', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: 123, sat_resolved_by: undefined },
    {sat_id:13, sat_title: 'My Works - Claimed Work - UQ:1494946 - uqmdeben', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: 23, sat_resolved_by: undefined },
    {sat_id:14, sat_title: '(UQ eSpace) Keywords with full stops at the end', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: undefined, sat_resolved_by: undefined },
    {sat_id:51, sat_title: 'My Works - Disowned Work - UQ:4128e45 - uqsmcor3', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: 444, sat_resolved_by: undefined },
    {sat_id:16, sat_title: 'My Works - Claimed Work - UQ:e9a2910 - uqnpacha', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: undefined, sat_resolved_by: undefined },
    {sat_id:71, sat_title: 'My Works - Disowned Work - UQ:02daQbe - uqrelli6', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: undefined, sat_resolved_by: undefined },
    {sat_id:188, sat_title: 'My Works - Claimed Work - UQ:4489ca8 - uqwtomas', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: 2333, sat_resolved_by: undefined },
];

export const adminDashboardReportWorksData = [
    {
        id: 1,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
    {
        id: 2,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
    {
        id: 3,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
    {
        id: 4,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
    {
        id: 5,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
    {
        id: 6,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
];

export const adminDashboardReportSystemAlertsData = [
{
    id: 1,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 2,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 3,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 4,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 5,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 6,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 7,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
}];