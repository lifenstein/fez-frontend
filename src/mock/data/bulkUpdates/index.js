export { default as bulkUpdatesList } from './bulkUpdatesList';