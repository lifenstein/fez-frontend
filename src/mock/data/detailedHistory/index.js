export { default as detailedHistory } from './detailedHistoryList';