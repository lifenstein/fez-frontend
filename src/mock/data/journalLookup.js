export const journalLookup = {
    total: "3",
    data: [
        {
            "jnl_jid": 1,
            "jnl_title": "Abstract and Applied Analysis",
            "jnl_publisher": "Publisher",
            "jnl_created_date": "2010-11-19",
            "jnl_updated_date": "2010-11-19",
            "fez_journal_issn": [
                {
                    "jnl_issn_jid": 1,
                    "jnl_issn": "1085-3375",
                    "jnl_issn_order": 1,
                },
                {
                    "jnl_issn_jid": 1,
                    "jnl_issn": "1687-0409",
                    "jnl_issn_order": 2,
                },
            ],
        },
        {
            "jnl_jid": 2,
            "jnl_title": "Acta Mathematica Vietnamica",
            "jnl_publisher": "Publisher 2",
            "jnl_created_date": "2011-01-19",
            "jnl_updated_date": "2011-01-19",
            "fez_journal_issn": [
                {
                    "jnl_issn_jid": 2,
                    "jnl_issn": "0251-4184",
                    "jnl_issn_order": 1,
                },
            ],
        },
        {
            "jnl_jid": 3,
            "jnl_title": "Journal of Physics D: Applied Physics",
            "jnl_publisher": "Publisher 1",
            "jnl_created_date": "2011-01-19",
            "jnl_updated_date": "2011-01-19",
            "fez_journal_issn": [
                {
                    "jnl_issn_jid": 3,
                    "jnl_issn": "0251-4184",
                    "jnl_issn_order": 1,
                },
                {
                    "jnl_issn_jid": 3,
                    "jnl_issn": "1361-6463",
                    "jnl_issn_order": 1,
                },
            ],
        },
    ],
};