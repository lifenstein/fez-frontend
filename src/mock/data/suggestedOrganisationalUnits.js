export const suggestedOrganisationalUnits = {
    "data": [
        {
            "aut_id": 7624839,
            "org_title": "Information Systems and Resource Services (University of Queensland Library)",
            "org_id": 1248
        }
    ]
};