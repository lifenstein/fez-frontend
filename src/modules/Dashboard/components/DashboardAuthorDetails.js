import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';
import withStyles from '@mui/styles/withStyles';

export const styles = theme => ({
    authorDetails: {
        color: theme.palette.white.main,
    },
});

export class DashboardAuthorDetails extends PureComponent {
    static propTypes = {
        title: PropTypes.string,
        familyName: PropTypes.string,
        givenName: PropTypes.string,
        orgUnits: PropTypes.array,
        positions: PropTypes.array,
        classes: PropTypes.object,
    };

    constructor(props) {
        super(props);
    }

    isCasualPosition(position) {
        return position.indexOf('Casual ') > -1;
    }

    /**
     * Hide roles that include the word 'Casual', where multiple roles are currently displayed,
     * and at least one does not include the word 'Casual'
     * If all appointments include the word 'Casual' we will need to display all roles.
     */
    areAllCasualPositions(positions) {
        if (!positions || positions.length === 0) {
            return false;
        }

        return (
            positions.filter(position => {
                return this.isCasualPosition(position);
            }).length === positions.length
        );
    }

    render() {
        const { classes } = this.props;
        const areAllCasualPositions = this.areAllCasualPositions(this.props.positions);
        return (
            <Grid container spacing={1}>
                <Grid item xs={12}>
                    <Typography variant={'h6'} className={classes.authorDetails}>
                        {this.props.title}&nbsp;{this.props.givenName}&nbsp;{this.props.familyName}
                    </Typography>
                </Grid>
                {/* Author Name/Positions/OrgUnits */}
                <Grid item xs={12}>
                    {this.props.positions &&
                        this.props.positions.length > 0 &&
                        this.props.positions.map(
                            (item, index) =>
                                ((!areAllCasualPositions && !this.isCasualPosition(item)) || areAllCasualPositions) && (
                                    <Typography
                                        key={index}
                                        variant={'caption'}
                                        component={'span'}
                                        className={classes.authorDetails}
                                    >
                                        <b>{item}</b>
                                        {this.props.orgUnits &&
                                            this.props.orgUnits.length > 0 &&
                                            this.props.orgUnits[index] &&
                                            `, ${this.props.orgUnits[index]}`}
                                    </Typography>
                                ),
                        )}
                </Grid>
            </Grid>
        );
    }
}

export default withStyles(styles, { withTheme: true })(DashboardAuthorDetails);
