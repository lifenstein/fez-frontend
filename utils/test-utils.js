/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { Route } from 'react-router';
import { mui1theme } from 'config/theme';
import { Provider } from 'react-redux';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';

import { getStore } from '../src/config/store';
import Immutable from 'immutable';
import { createMemoryHistory } from 'history';

const domTestingLib = require('@testing-library/dom');
const reactTestingLib = require('@testing-library/react');
const mime = require('mime-types');

const { configure } = domTestingLib;

configure(config => ({
    ...config,
    testIdAttribute: 'id',
}));

export const AllTheProviders = props => {
    return (
        <MuiThemeProvider theme={mui1theme}>
            <MuiPickersUtilsProvider utils={MomentUtils}>{props.children}</MuiPickersUtilsProvider>
        </MuiThemeProvider>
    );
};

AllTheProviders.propTypes = {
    children: PropTypes.node,
};

export const rtlRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }), renderMethod = render } = {},
) => {
    return renderMethod(
        <AllTheProviders>
            <Router history={history}>{ui}</Router>
        </AllTheProviders>,
    );
};

export const WithRouter = ({ children, route = '/', history = createMemoryHistory({ initialEntries: [route] }) }) => (
    <AllTheProviders>
        <Router history={history}>{children}</Router>
    </AllTheProviders>
);
export const renderWithRedux = ({ initialState }) => render => {
    return {
        ...render,
        store: getStore({ initialState, history: render.history }),
    };
};

export const withRouter = ({
    route = '/',
    path = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
} = {}) => WrappedComponent => {
    return (
        <Router history={history}>
            <Route path={path} children={WrappedComponent} />
        </Router>
    );
};

export const withRedux = (initialState = Immutable.Map()) => WrappedComponent => {
    return <Provider store={getStore(initialState)}>{WrappedComponent}</Provider>;
};

export const WithReduxStore = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>
        <AllTheProviders>{children}</AllTheProviders>
    </Provider>
);

export const defaultDatastream = {
    dsi_embargo_date: null,
    dsi_open_access: 1,
    dsi_label: '',
    dsi_copyright: null,
    dsi_state: 'A',
    dsi_size: 27932352,
};
export const createFezDatastreamInfoArray = (datastreams, pid = null, withPreview = true) => {
    let processed = datastreams.filter(datastream => typeof datastream === 'object' || typeof datastream === 'string');
    if (withPreview) {
        processed = [];
        datastreams.forEach(function(datastream) {
            processed.push(datastream);
            const filename = typeof datastream === 'object' ? datastream.dsi_dsid : datastream;
            const mimetype = mime.lookup(filename);
            // bail in case it's not derivable
            if (
                !filename.includes('.pdf') &&
                !mimetype.includes('image') &&
                !mimetype.includes('audio') &&
                !mimetype.includes('video')
            ) {
                return;
            }
            // eslint-disable-next-line no-nested-ternary
            const derivativeMimetype = mimetype.includes('audio')
                ? 'audio/mp3'
                : mimetype.includes('video')
                ? 'video/mp4'
                : 'image/jpg';

            (derivativeMimetype.includes('image') ? ['preview_', 'thumbnail_', 'web_'] : ['']).forEach(function(
                prefix,
            ) {
                processed.push({
                    dsi_dsid: `${prefix}${filename.split('.')[0]}_t.${derivativeMimetype.split('/').pop()}`,
                    dsi_mimetype: derivativeMimetype,
                });
            });
        });
    }

    return processed.map((datastream, index) => ({
        ...defaultDatastream,
        ...{
            dsi_pid: pid,
            dsi_order: index + 1,
            dsi_mimetype: mime.lookup(typeof datastream === 'object' ? datastream.dsi_dsid : datastream),
        },
        ...(typeof datastream === 'object'
            ? datastream
            : {
                  dsi_dsid: datastream,
              }),
    }));
};

export const withDatastreams = (sources, datastreams, callback) =>
    datastreams.map(item => {
        const source = sources.find(datastream =>
            datastream.dsi_dsid.includes(
                item.dsi_dsid
                    .replace('_t.', '.')
                    .replace('preview_', '')
                    .replace('thumbnail_', '')
                    .replace('web_', '')
                    .split('.')[0],
            ),
        );
        const isDerivative = item.dsi_dsid !== source.dsi_dsid;
        callback(item, source, isDerivative);
    });

export const getDatastreamByFilename = (filename, derivatives) =>
    derivatives.find(derivative => derivative.dsi_dsid === filename);

module.exports = {
    ...domTestingLib,
    ...reactTestingLib,
    rtlRender,
    renderWithRouter,
    renderWithRedux,
    withRedux,
    withRouter,
    AllTheProviders,
    WithReduxStore,
    WithRouter,
    createFezDatastreamInfoArray,
    getDatastreamByFilename,
    withDatastreams,
};
