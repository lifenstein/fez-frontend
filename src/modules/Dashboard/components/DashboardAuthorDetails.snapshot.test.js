import React from 'react';
import { DashboardAuthorDetails } from './DashboardAuthorDetails';
import { currentAuthor, authorDetails } from 'mock/data';
import { rtlRender } from 'test-utils';

function setup(testProps = {}) {
    const props = {
        classes: {},
        ...testProps,
    };
    return rtlRender(<DashboardAuthorDetails {...props} />);
}

describe('Dashboard Author Details test', () => {
    it('Render the authors details as expected for a UQ researcher)', () => {
        const { container } = setup({
            title: currentAuthor.uqresearcher.aut_title || '',
            givenName: currentAuthor.uqresearcher.aut_fname || '',
            familyName: currentAuthor.uqresearcher.aut_lname || '',
            orgUnits: authorDetails.uqresearcher.org_units,
            positions: authorDetails.uqresearcher.positions,
        });
        expect(container).toMatchSnapshot();
    });
});
