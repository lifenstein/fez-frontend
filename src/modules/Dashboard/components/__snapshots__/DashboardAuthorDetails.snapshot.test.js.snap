// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Author Details test Render the authors details as expected for a UQ researcher) 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1uywuy6-MuiTypography-root"
      >
         
         
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-1e4ju4k-MuiTypography-root"
      >
        <b>
          Affiliate Professor
        </b>
        , Institute for Molecular Bioscience
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption css-1e4ju4k-MuiTypography-root"
      >
        <b>
          Affiliate Professorial Res Fellow
        </b>
        , School of Chemistry and Molecular Biosciences
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption css-1e4ju4k-MuiTypography-root"
      >
        <b>
          ARC Australian Laureate Fellow
        </b>
        , The University of Queensland Diamantina Institute
      </span>
    </div>
  </div>
</div>
`;
